chainercv/links/model/ssd/multibox_coder.py

```diff
@@ -0,0 +1,231 @@
+        index[:] = -1
+
+        masked_iou = iou.copy()
+        while True:
+            i, j = np.unravel_index(masked_iou.argmax(), masked_iou.shape)
+            if masked_iou[i, j] <= 1e-6:
+                break
+            index[i] = j
+            masked_iou[i, :] = 0
+            masked_iou[:, j] = 0
+
+        mask = xp.logical_and(index < 0, iou.max(axis=1) >= iou_thresh)
+        index[mask] = iou[mask].argmax(axis=1)
+
+        mb_bbox = bbox[index]
+        mb_loc = xp.hstack((
+            ((mb_bbox[:, :2] + mb_bbox[:, 2:]) / 2
+             - self._default_bbox[:, :2]) /
+            (self._variance[0] * self._default_bbox[:, 2:]),
+            xp.log((mb_bbox[:, 2:] - mb_bbox[:, :2])
+                   / self._default_bbox[:, 2:]) /
+            self._variance[1]))
+
+        # [0, n_fg_class - 1] -> [1, n_fg_class]
+        mb_label = label[index] + 1
+        # 0 is for background
+        mb_label[index < 0] = 0
+
+        return mb_loc.astype(np.float32), mb_label.astype(np.int32)
+
+    def decode(self, mb_loc, mb_conf, nms_thresh, score_thresh):
+        """Decodes coordinates and classes of bounding boxes.
+
+        This method decodes :obj:`mb_loc` and :obj:`mb_conf` returned
+        by a SSD network.
+
+        Args:
+            mb_loc (array): A float array whose shape is
+                :math:`(K, 4)`, :math:`K` is the number of
+                 default bounding boxes.
+            mb_conf (array): A float array whose shape is
+                :math:`(K, n\_fg\_class + 1)`.
+            nms_thresh (float): The threshold value
+                for :meth:`chainercv.transfroms.non_maximum_suppression`.
+            score_thresh (float): The threshold value for confidence score.
+                If a bounding box whose confidence score is lower than
+                this value, the bounding box will be suppressed.
+
+        Returns:
+            tuple of three arrays:
+            This method returns a tuple of three arrays,
+            :obj:`(bbox, label, score)`.
+
+            * **bbox**: A float array of shape :math:`(R, 4)`, \
+                where :math:`R` is the number of bounding boxes in a image. \
+                Each bouding box is organized by \
+                :obj:`(x_min, y_min, x_max, y_max)` \
+                in the second axis.
+            * **label** : An integer array of shape :math:`(R,)`. \
+                Each value indicates the class of the bounding box.
+            * **score** : A float array of shape :math:`(R,)`. \
+                Each value indicates how confident the prediction is.
+
+        """
+        xp = self.xp
+
+        # the format of raw_bbox is (center_y, center_x, height, width)
+        raw_bbox = xp.hstack((
+            self._default_bbox[:, :2] + mb_loc[:, :2] *
+            self._variance[0] * self._default_bbox[:, 2:],
+            self._default_bbox[:, 2:] *
+            xp.exp(mb_loc[:, 2:] * self._variance[1])))
```

Comment:
It's a matter of taste, but I think this is more readable with shorter code and explicit variable names.

```python
center = self._default_bbox[:, :2] + mb_loc[:, :2] * self._variance[0] * self._default_bbox[:, 2:],
size = self._default_bbox[:, 2:] * xp.exp(mb_loc[:, 2:] * self._variance[1])

raw_bbox = xp.hstack((center - size / 2, center + size / 2)
```